course_discovery/apps/course_metadata/management/commands/publish_live_course_runs.py

```
@@ -0,0 +1,38 @@
+import datetime
+import logging
+
+import pytz
+from django.core.management import BaseCommand, CommandError
+from django.utils.translation import ugettext as _
+
+from course_discovery.apps.course_metadata.choices import CourseRunStatus
+from course_discovery.apps.course_metadata.models import CourseRun
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = 'Based on course run go_live dates, publish any runs that are due'
+
+    def handle(self, *args, **options):
+        now = datetime.datetime.now(pytz.UTC)
+        failed = False
+
+        for course_run in CourseRun.objects.all():
+            reviewed = course_run.status == CourseRunStatus.Reviewed
+            past_due = course_run.go_live_date and course_run.go_live_date <= now
+            if not reviewed or not past_due:
+                continue
+
+            logger.info(_('Publishing course run {key}').format(key=course_run.key))
+
+            try:
+                course_run.publish()
+            except Exception:  # pylint: disable=broad-except
```

nit: I am still not a fan of Bare exceptions. Any chance we can be more specific here?

---

Minor nitpick - "bare" exceptions and "broad" exceptions are different. "Bare" is without the word "Exception" in there, and I agree. Those are 99% of the time a bad idea.

But I'd push back on broad exceptions being always bad. I believe they are useful in situations like this where you are treating a code block like a black box and don't care why something fails. You just want to ensure that any failure doesn't stop the loop.

Like... even if we had a list of all possible errors that could be raised, what would be the deciding factor between continuing the loop and stopping it? I guess an error that would suggest the rest of the publish attempts would fail anyway, so why bother. But I'm not sure what those are.

---

I guess a common complaint of broad exceptions is also that it makes it hard to debug, because they hide errors. Which is fair if the code is trying to "handle" an error silently. But for cases where you catch-and-log, it seems a little less bad to me.

---

I think my distaste comes from what you mentioned.  Catch and log seems reasonable.